app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -0,0 +1,128 @@
+class Api::V1::BookmarksController < Api::V1::BaseController
+  respond_to :json
+
+  def create
+    archivist = User.find_by_login(params[:archivist])
+    external_bookmarks = params[:bookmarks]
```

Comment:
I'm not sure it would work for the other types as is, because the `external_bookmark` method in the API `bookmark_controller` coerces the JSON input into a structure with an `external` hash in it, which IIRC is the structure passed back by the normal `bookmark_controller` when creating a bookmark for an external work. I think this brings the `external=` virtual attribute in the `Bookmark` model into play. 

Having said that, I didn't inspect the other bookmark types, and it's been a while since I unpicked how bookmarks are created, so you might well be right. I'll rename that variable to make it less external-centric which will make it easier to adapt this if we have a use case for creating other types of bookmarks in future.